plugins/editor/index.js

```diff
@@ -20,51 +20,24 @@ function editor(app, opts, done){
 
   var codeEditor;
   var outputConsole;
-  var buffer = new ConsoleBuffer();
-
-  var refreshQueued = null;
-  var lastRefresh = 0;
-  var refreshDelayMillis = 64;
-
-  var space = app.workspace;
-
-  function output(evt){
-    buffer.update(evt);
-
-    if(refreshQueued != null){
-      return;
-    }
-    if(lastRefresh < Date.now() - refreshDelayMillis){
-      refreshConsole();
-    }else{
-      refreshQueued = setTimeout(refreshConsole, refreshDelayMillis);
-    }
-  }
+  var state = {};
 
   function refreshConsole(){
+    const { buffer } = state;
     if(outputConsole){
       outputConsole.innerHTML = buffer.getConsoleHTML();
       outputConsole.scrollTop = outputConsole.scrollHeight;
     }
-    refreshQueued = null;
-    lastRefresh = Date.now();
   }
 
-  function clearOutput(){
-    buffer.clear();
-    if(outputConsole){
-      outputConsole.innerHTML = buffer.getConsoleHTML();
-    }
-    if(refreshQueued != null){
-      clearInterval(refreshQueued);
-      refreshQueued = null;
-    }
-    lastRefresh = 0;
+  function onChange() {
```

Comment:
why not combine `refreshConsole` and `onChange`?  seems unnecessary to have both